backend/molgenis-emx2-sql/src/test/java/org/molgenis/emx2/sql/SqlQueryBuilderHelpersTest.java

```diff
@@ -0,0 +1,178 @@
+    assertEquals(
+        "select * from \"SqlQueryBuilderHelpersTest\".\"testTable\" order by \"testColumn\" asc",
+        ascQuery.getSQL());
+  }
+
+  @Test
+  void orderByDESC() {
+    final SqlTableMetadata tableMetadata = testTable.getMetadata();
+    final DSLContext jooq = tableMetadata.getJooq();
+
+    SelectJoinStep<org.jooq.Record> from = jooq.select().from(tableMetadata.getJooqTable());
+
+    final SelectColumn select = new SelectColumn("testColumn");
+    select.setOrderBy(Collections.singletonMap("testColumn", Order.DESC));
+    SelectConnectByStep<Record> ascQuery =
+        SqlQueryBuilderHelpers.orderBy(tableMetadata, select, from);
+    assertEquals(
+        "select * from \"SqlQueryBuilderHelpersTest\".\"testTable\" order by \"testColumn\" desc",
+        ascQuery.getSQL());
+  }
+
+  @Test
+  void orderByRef() {
+    testSchema.create(new TableMetadata("refTable").add(new Column("refColumn").setPkey()));
+    final SqlTableMetadata tableMetadata = testTable.getMetadata();
+    tableMetadata.add(new Column("refColumn").setType(ColumnType.REF).setRefTable("refTable"));
+    final DSLContext jooq = tableMetadata.getJooq();
+
+    final SelectColumn select = new SelectColumn("refColumn");
+    select.setOrderBy(Collections.singletonMap("refColumn", Order.ASC));
+    SelectConnectByStep<Record> ascQuery =
+        SqlQueryBuilderHelpers.orderBy(
+            tableMetadata, select, jooq.select().from(tableMetadata.getJooqTable()));
+    assertEquals(
+        "select * from \"SqlQueryBuilderHelpersTest\".\"testTable\" order by \"refColumn\" asc",
+        ascQuery.getSQL());
+
+    final SelectColumn selectDesc = new SelectColumn("refColumn");
+    selectDesc.setOrderBy(Collections.singletonMap("refColumn", Order.DESC));
+    SelectConnectByStep<Record> descQuery =
+        SqlQueryBuilderHelpers.orderBy(
+            tableMetadata, selectDesc, jooq.select().from(tableMetadata.getJooqTable()));
+    assertEquals(
+        "select * from \"SqlQueryBuilderHelpersTest\".\"testTable\" order by \"refColumn\" desc",
+        descQuery.getSQL());
+  }
+
+  @Test
+  void getNonExistingColumn() {
+    final TableMetadata table = new TableMetadata("testTable");
+    assertThrows(
+        MolgenisException.class, () -> SqlQueryBuilderHelpers.getColumnByName(table, "foo"));
+  }
+
+  @Test
+  void getSearchColumn() {
+    final TableMetadata table = new TableMetadata("testTable");
+    final Column column = SqlQueryBuilderHelpers.getColumnByName(table, TEXT_SEARCH_COLUMN_NAME);
+    assertEquals("testTable" + TEXT_SEARCH_COLUMN_NAME, column.getName());
+  }
+
+  @Test
+  void getColumnByName() {
+    final TableMetadata table = new TableMetadata("testTable");
+    table.add(new Column("col1"));
+    final Column column = SqlQueryBuilderHelpers.getColumnByName(table, "col1");
+    assertEquals("col1", column.getName());
+  }
+
+  @Test
+  void getRefColumnByName() {
+    final TableMetadata table = Mockito.mock(TableMetadata.class);
```

Comment:
why do you use a mock here? Why not simply create a new TableMetadata()? Because that is just a stupid bean so mocking seems to be more complex than the real thing?